packages/codeeditor/src/editor.ts

```diff
@@ -543,6 +538,61 @@ namespace CodeEditor {
   type Factory = (options: IOptions) => CodeEditor.IEditor;
 
   /**
+   * The configuration options for an editor.
+   */
+  export
+  interface IConfig {
+    /**
+     * Whether line numbers should be displayed.
+     */
+    lineNumbers: boolean;
+
+    /**
+     * Set to false for horizontal scrolling.
+     */
+    wordWrap: boolean;
+
+    /**
+     * Whether the editor is read-only.
+     */
+    readOnly: boolean;
+
+    /**
+     * The number of spaces a tab is equal to.
+     */
+    tabSize: number;
+
+    /**
+     * Whether to insert spaces when pressing Tab.
+     */
+    insertSpaces: boolean;
+
+    /**
+     * Whether to highlight matching brackets when one of them is selected.
+     */
+    matchBrackets: boolean;
+
+    /**
+     * Whether to automatically close brackets after opening them.
+     */
+    autoClosingBrackets: boolean;
+  }
+
+  /**
+   * The default configuration options for an editor.
+   */
+  export
+  let defaultConfig: IConfig = {
+    lineNumbers: false,
+    wordWrap: true,
+    readOnly: false,
+    tabSize: 4,
+    insertSpaces: true,
```

Comment:
Hmm, I am seeing a mix of tabs in spaces in the standalone editor. When it autoindents, I get spaces, but if I just tab over I get tabs. See attached.

![screen shot 2017-06-19 at 3 48 05 pm](https://user-images.githubusercontent.com/27600/27309055-c079e8ac-5506-11e7-9edb-c2ec8ccc51ff.png)